src/clients/googleAnalytics.ts

```diff
@@ -0,0 +1,29 @@
+type EventParameters = Parameters<(eventName: string, parameters?: Record<string, string>) => void>;
+
+const googleAnalytics = () => {
+  let gtag: Gtag.Gtag;
+  const loadGTag = (callback: (...args: EventParameters) => void) => (...args: EventParameters) => {
+    gtag = window?.gtag;
+    if (!gtag) {
+      if (process.env.NODE_ENV !== 'production') {
+        console.warn('gtag not loaded. Google Analytics event not sent', args);
+      }
+    } else {
+      callback(...args);
+    }
+  };
```

Comment:
You can avoid nesting IF statements:
```typescript
if (gtag) {
  callback(...args);
} else if (process.env.NODE_ENV !== 'production') {
  console.warn('gtag not loaded. Google Analytics event not sent', args);
}
```

Reply:
I don't want to do that because I want to check if the tag exists or I want to call the callback. I only want to log errors in non-production environments. We could decide we want to handle the non existence of gtag another way but I though  would add this as a minimum

Reply:
That's what this code does: it only logs in non-production environments, if `gtag` doesn't exist (otherwise it will execute the callback).

It's not a major thing, so if you prefer your syntax I'm happy to approve the PR. I thought I'd just highlight you can avoid the nesting 🙂

Reply:
you are right I misread it 🙃